sbt-launcher/src/main/scala/coursier/sbtlauncher/MainApp.scala

```diff
@@ -1,363 +1,22 @@
-          .lastOption
-          .map(_._1)
-      } else
-        None
-
-    val foundSbtLmCoursierVersionOpt =
-      if (!config.sbtVersion.startsWith("0.") && options.addCoursier) {
-
-        val fromProject = sbtCoursierVersionFromPluginsSbt(new File(sys.props("user.dir") + "/project"), "sbt-lm-coursier")
-        val global = sbtCoursierVersionFromPluginsSbt(new File(sys.props("sbt.global.base") + "/plugins"), "sbt-lm-coursier")
-
-        (fromProject.toSeq ++ global.toSeq)
-          .sortBy { case (v, _) => coursier.core.Version(v) }
-          .lastOption
-          .map(_._1)
-      } else
-        None
-
-    val (sbtCoursierVersionOpt, sbtLmCoursierVersionOpt) =
-      if (options.addCoursier)
-        (foundSbtCoursierVersionOpt, foundSbtLmCoursierVersionOpt) match {
-          case (Some(v), _) =>
-            (Some(v), None)
-          case (None, Some(v)) =>
-            (None, Some(v))
-          case (None, None) =>
-            if (config.sbtVersion.startsWith("0.13."))
-              (Some("1.1.0-M7"), None) // last sbt 0.13 compatible version
-            else
-              // now adding sbt-lm-coursier by default, that adds a coursier-based DependencyResolution in particular
-              options.coursierPlugin.getOrElse("sbt-lm-coursier") match {
-                case "sbt-coursier" =>
-                  (Some(Properties.sbtCoursierDefaultVersion), None)
-                case "sbt-lm-coursier" =>
-                  (None, Some(Properties.sbtCoursierDefaultVersion))
-                case other =>
-                  System.err.println(s"Unrecognized coursier plugin: $other")
-                  sys.exit(1)
-              }
-        }
-      else
-        (None, None)
-
-    // TODO Warn if both sbt-coursier and sbt-lm-coursier are found?
-
-    val appId = ApplicationID(
-      config.organization,
-      config.moduleName,
-      config.version,
-      config.mainClass,
-      options.mainComponents.toArray,
-      crossVersioned = false,
-      xsbti.CrossValue.Disabled,
-      options.classpathExtra.map(new File(_)).toArray
-    )
-
-    doRun(
-      appId,
-      remainingArgs.all.toArray,
-      RunParams(
-        config.scalaVersion,
-        config.sbtVersion,
-        config.sbtBinaryVersion,
-        addSbtLauncherPlugin = sbtCoursierVersionOpt.exists(v => Version(v).compare(Version("1.1.0-M12")) < 0),
-        sbtCoursierVersionOpt,
-        sbtLmCoursierVersionOpt,
-        config.dependencies ++ extraDeps
+import java.util.Locale
+
+import caseapp.core.RemainingArgs
+
+object MainApp {
+  def main(args: Array[String]): Unit = {
+    val fromEnv = sys.env.get("COURSIER_SBT_LAUNCHER_PARSE_ARGS")
+      .map(s => s == "1" || s.toLowerCase(Locale.ROOT) == "true")
+    def fromProps = sys.props.get("coursier.sbt-launcher.parse-args")
+      .map(s => s == "1" || s.toLowerCase(Locale.ROOT) == "true")
+    val parseArgs = fromEnv.orElse(fromProps).getOrElse(false)
```

Comment:
(minor) shouldn't sys props trump env vars?

Reply:
I've been doing the opposite in coursier. That allows users to use the environment to override whatever a script has set itself via a Java property.

Reply:
Why can't a user override with a system property or with a command line option?

Reply:
That would work with sbt-extras, yes. But not from more convoluted scripts, that call coursier, or sbt-extras, on their own. (Although I guess it's even for that particular setting…)

Reply:
Yeah, I see the use case.  But I think it's backwards (and a common pitfall - sbt has a bunch of hidden sys prop usages).

It's probably best to go with this for now, and maybe later consider changing it holistically.